File: javascript/merge_sort/src/merge_sort.js
const sort = (arr) => {
    if (arr.length < 2) {
        return arr;
    } else {
        return merge(sort(arr.slice(0, arr.length / 2)), sort(arr.slice(arr.length / 2, arr.length)));
    }
}

const merge = (arr1, arr2) => {
    merged_arr = [];
    let i = 0, j = 0;
    while(merged_arr.length < arr1.length + arr2.length) {
        merged_arr.push(i >= arr1.length || (j < arr2.length && arr2[j] < arr1[i])? arr2[j++] : arr1[i++]);
    }
    return merged_arr;
}

exports.sort = sort;

File: python/merge_sort/merge_sort.py
def sort(lst):
    if len(lst) < 2:
        return lst
    else:
        return merge(sort(lst[0:len(lst) // 2]), sort(lst[(len(lst) // 2) : len(lst)]))

def merge(lst1, lst2):
    merged = []
    i = 0
    j = 0
    while len(merged) < len(lst1) + len(lst2):
        if i >= len(lst1) or (j < len(lst2) and lst2[j] < lst1[i]):
            merged.append(lst2[j])
            j += 1
        else:
            merged.append(lst1[i])
            i += 1
    return merged
